dlt-connector/src/controller/Community.ts

```diff
@@ -0,0 +1,36 @@
+import { CommunityDraft } from '@/graphql/input/CommunityDraft'
+import { uuid4ToBuffer } from '@/utils'
+import { Community } from '@entity/Community'
+import { getDataSource } from '@typeorm/DataSource'
+import { crypto_generichash as cryptoHash } from 'sodium-native'
+
+export const iotaTopicFromCommunityUUID = (communityUUID: string): string => {
+  const hash = Buffer.alloc(32)
+  cryptoHash(hash, uuid4ToBuffer(communityUUID))
+  return hash.toString('hex')
+}
+
+export const isExist = async (community: CommunityDraft | string): Promise<boolean> => {
+  const iotaTopic =
+    community instanceof CommunityDraft ? iotaTopicFromCommunityUUID(community.uuid) : community
+  const result = await getDataSource().manager.find(Community, {
+    where: { iotaTopic },
+  })
+  return result.length > 0
+}
+
+export const create = (community: CommunityDraft, topic?: string): Community => {
+  const communityEntity = new Community()
+  communityEntity.iotaTopic = topic ?? iotaTopicFromCommunityUUID(community.uuid)
+  communityEntity.createdAt = new Date(community.createdAt)
```

Comment:
the column `createdAt` annotated with `@CreateDateColumn` will set its value automatically. So please do not set it manually in the code. I don't know if the given value or the automatical value will be stored in database.

Reply:
Oh ok, I thought I must use it together with default: TIMESTAMP.
I have it replaced with Column.